frontend/src/utils/isSwear.js

```diff
@@ -0,0 +1,23 @@
+import list from './badWords.json';
+
+const badWords = list.words;
+
+function escapeRegExp(string) {
+	// eslint-disable-next-line no-useless-escape
+	return string.replace(/([.*+?^=!:${}()|\[\]\/\\])/g, '\\$1');
+}
+
+export default function isSwear(query) {
+	var badWordsPresent = badWords.filter((badWord) => {
+		var regex = '\\b';
+		regex += escapeRegExp(badWord);
+		regex += '\\b';
+		return new RegExp(regex, 'i').test(query);
+	});
+
+	if (badWordsPresent.length > 0) {
+		return true;
+	} else {
+		return false;
+	}
+}
```

Comment:
This can be improved by iterating the list of bad words and breaking the iteration after hitting one word. Currently this function will iterate the whole list even if the first word of the list is present in the string.

Reply:
As a tip, you could also drop the `ìf` and `else` statements and just `return badWordsPresent.length > 0`.